### PR TITLE
Added leading zeros to RU numbers in rack elevations

### LINKED ISSUE #5274: Add the ability for RU numbers to display leading zeros
### As ...

Nelly - Network Engineer

### I want ...

Rack units to have leading zeros in the Rack Detail View

### So that ...

The elevation view better represents the real-world label that is on my rack

### I know this is done when...

Rack units labels will display 01, 02, 03, ... instead of 1, 2, 3, ...

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [X] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [X] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

### Database Changes

Should not require a DB change, just display format.

### External Dependencies

None

### PR DESCRIPTION
**Pull request by tsm1th**
# Closes #5274 
# What's Changed

    - A new configuration setting for rack elevations which displays a minimum of two-digits (01,02,03.. instead of 1,2,3..) for rack unit numbers.  
    - UI/UX change only.
# Screenshots
After | Before
----- | ------ |
Screenshot 2024-03-29 at 2 23 28 PM | Screenshot 2024-03-29 at 9 53 25 AM|

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Example App Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/5274.changed [hunk 1]
```diff
@@ -0,0 +1 @@
+Changed RU numbers to display a minimum of two digits in rack elevations.
```
Comment (HanlinMiao) [inline changes/5274.changed:1]: Should we do this to make the changelog more explicit?
```suggestion
Changed Reserved Unit numbers to display a minimum of two digits in rack elevations.
```

**Review by HanlinMiao — APPROVED** [review]

**Comment by abates** [comment]
For what it's worth, I have a rack here in my office that does not have leading zeros, so this change would actually make Nautobot less representative of my rack labeling. I'm wondering if this should be configurable?

**Comment by tsm1th** [comment]
> For what it's worth, I have a rack here in my office that does not have leading zeros, so this change would actually make Nautobot less representative of my rack labeling. I'm wondering if this should be configurable?

@abates I thought I had seen it both ways as well.  There is a boolean field for `descending units` on a per rack basis, so maybe this should be a boolean as well. 🤔

**Comment by glennmatthews** [comment]
Agreed, this needs to be configurable. IMHO making it a global configuration setting may be sufficient and simpler than adding it as a per-rack toggle?

**Comment by tsm1th** [comment]
@glennmatthews I think that would suffice unless a user has both rack types in their environment.  They might want to distinguish between them like they can do with the `descending units` attribute?

**Comment by itdependsnetworks** [comment]
my 2 cents :) 

I don't disagree that they could have both, but imho there will not be confusion if your work order said "RU 1" and it read "RU 01" as an example. Given the effort, a global setting seems to be more that enough options.

**Comment by tsm1th** [comment]
I would agree given the effort.  Are we thinking something like the following? 

```python
unit_display = f"{unit:02d}" if settings.UI_RACK_UNIT_TWO_DIGIT_FORMAT else str(unit)
drawing.add(drawing.text(unit_display, position_coordinates, class_="unit"))
```

**Comment by glennmatthews** [comment]
Yes, though I'd suggest something along the lines of `RACK_ELEVATION_UNITS_TWO_DIGIT_FORMAT` just to more closely match the existing `RACK_ELEVATION_DEFAULT_UNIT_HEIGHT` and `RACK_ELEVATION_DEFAULT_UNIT_WIDTH` setting.

**Comment by tsm1th** [comment]
@glennmatthews Following the same way those settings are implemented, it's now a configuration parameter in constance or a settings override. 

Screenshot 2024-04-01 at 5 03 41 PM

**Review by glennmatthews — APPROVED** [review]
Good stuff!

File: nautobot/core/templates/nautobot_config.py.j2 [hunk 2]
```diff
@@ -472,6 +472,13 @@ INSTALLATION_METRICS_ENABLED = is_truthy(os.getenv("NAUTOBOT_INSTALLATION_METRIC
 # ):
 #     RACK_ELEVATION_DEFAULT_UNIT_WIDTH = int(os.environ["NAUTOBOT_RACK_ELEVATION_DEFAULT_UNIT_WIDTH"])
 
+# Enable two-digit format for the rack unit numbering in rack elevations.
+#if (
+#    "NAUTOBOT_RACK_ELEVATION_UNIT_TWO_DIGIT_FORMAT" in os.environ
+#     and os.environ["RACK_ELEVATION_UNIT_TWO_DIGIT_FORMAT"] != ""
+#):
+#    RACK_ELEVATION_UNIT_TWO_DIGIT_FORMAT = is_truthy(os.environ["RACK_ELEVATION_UNIT_TWO_DIGIT_FORMAT"])
```
[the remark this answers is not in the document]
Comment (tsm1th) [reply nautobot/core/templates/nautobot_config.py.j2:480]: Great catch!  Forgot to update the template once I found that typo.

**Pull request merged**